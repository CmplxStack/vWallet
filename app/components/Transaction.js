import React from 'react';
import moment from 'moment';

export default ({
  account,
  address,
  amount,
  blockhassh,
  category,
  confirmation,
  time,
  timereceived,
  txid
}) => (
  <div className="container">
    <div className="row">
      <h5>{moment.unix(timereceived).format('MM/DD/YYYY  hh:mm')}</h5>
    </div>
    <div className="row">
      <div className="col-md-2" style={{ color: 'white', fontWeight: 'bold' }}>
        Category:
      </div>
      {category.includes('receive') ? `Received from ${account}` : `Sent to ${address}`}
    </div>
    <div className="row">
      {' '}
      <div className="col-md-2" style={{ color: 'white', fontWeight: 'bold' }}>
        TXID:
      </div>{' '}
      {txid}
    </div>
  </div>
);
