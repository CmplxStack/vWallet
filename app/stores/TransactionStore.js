import { observable, action, computed } from 'mobx'

class TransactionStore {
  @observable transactions = []

  @action
  addTransaction = transaction => {
    this.transactions.push(transaction)
  }

  @action
  addTransactions = transactions => {
    this.transactions = [...this.transactions, ...transactions]
  }

  @computed
  get getTransactionCount() {
    return this.transaction.lenght
  }
}

const store = new TransactionStore()
export default store
