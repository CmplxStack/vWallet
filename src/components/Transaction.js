import React, { Component } from 'react'
import moment from 'moment'
import T from 'i18n-react'
import incoming from '../assets/images/incoming.png'
import outgoing from '../assets/images/outgoing.png'
import arrowdown from '../assets/images/arrowdown.png'
import { renderReporter, propTypes } from 'mobx-react'
import { shell } from 'electron'
import { isStringTextContainingNode } from 'typescript'
import styled from 'styled-components'
const XVGformatter = new Intl.NumberFormat('en-US', {
  style: 'decimal',
  minimumFractionDigits: 5,
})

const TextContainer = styled.div`
  color: ${props => (props.theme.light ? '#999999;' : '#7193ae;')};
`

export default class Transaction extends Component {
  constructor(props) {
    super(props)
    this.state = { hide: true }
  }

  render() {
    const {
      account,
      address,
      amount,
      blockhash,
      category,
      confirmations,
      time,
      timereceived,
      txid,
    } = this.props

    return (
      <div className="container">
        <div className="row">
          <div className="col-md-1">
            <TextContainer>
              {moment
                .unix(timereceived)
                .format('MMM')
                .toUpperCase()}
            </TextContainer>
            <TextContainer
              style={{
                fontWeight: '300',
                fontSize: '22px',
                lineHeight: '1.25',
              }}
            >
              {moment.unix(timereceived).format('DD')}
            </TextContainer>
          </div>
          <TextContainer
            className="col-md-2"
            style={{
              fontWeight: 'bold',
              marginTop: '8px',
            }}
          >
            {category.includes('receive') ? (
              <img src={incoming} />
            ) : (
              <img src={outgoing} />
            )}
          </TextContainer>
          <div
            className="col-md-7"
            style={{
              fontWeight: 'bold',
              color: category.includes('receive') ? '#00917a' : '#dc2b3d',
              textAlign: 'right',
              letterSpacing: '1px',
              fontSize: '22px',
            }}
          >
            <div>
              {category.includes('receive') ? '+' : ''}
              {amount.toFixed(2).toLocaleString('en-US')} XVG
            </div>
            <TextContainer
              style={{
                fontSize: '12px',
                fontWeight: '400',
                letterSpacing: '1px',
              }}
            >
              <font>
                {category.includes('receive')
                  ? T.translate('transaction.item.receive')
                  : T.translate('transaction.item.sent')}
              </font>
            </TextContainer>
          </div>
          {blockhash ? (
            <div
              className="col-md-2"
              style={{ textAlign: 'center', cursor: 'pointer' }}
              onClick={() => {
                this.setState({ hide: !this.state.hide })
              }}
            >
              <div>
                <img
                  src={arrowdown}
                  style={{
                    transform: `rotate(${this.state.hide ? 0 : 180}deg)`,
                  }}
                />
              </div>
              <TextContainer style={{ fontSize: '10px' }}>
                {this.state.hide
                  ? T.translate('transaction.item.details')
                  : T.translate('transaction.item.close')}
              </TextContainer>
            </div>
          ) : (
            <div
              className="col-md-2"
              style={{ textAlign: 'center', marginTop: '15px' }}
            >
              {' '}
              <TextContainer style={{ fontSize: '10px' }}>
                Out of sync
              </TextContainer>
            </div>
          )}
        </div>
        {!this.state.hide ? (
          <div className="trans-details">
            <div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.address')}:
              </div>{' '}
              <a
                className="col-md-10"
                href="#"
                onClick={() =>
                  shell.openExternal(
                    `https://verge-blockchain.info/address/${address}`
                  )
                }
              >
                {address}
              </a>
            </div>
            <div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.blockhash')}:
              </div>
              <div className="col-md-10">
                {' '}
                <a
                  href="#"
                  onClick={() =>
                    shell.openExternal(
                      `https://verge-blockchain.info/block/${blockhash}`
                    )
                  }
                >
                  <em>{blockhash.substring(0, 40)}...</em>
                </a>
              </div>
            </div>
            <div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.txid')}:
              </div>
              <div className="col-md-10">
                {' '}
                <a
                  href="#"
                  onClick={() =>
                    shell.openExternal(
                      `https://verge-blockchain.info/tx/${txid}`
                    )
                  }
                >
                  <em>{txid.substring(0, 40)}...</em>
                </a>
              </div>
            </div>
            {/* Sub division of the table */}
            <div className="row" style={{ marginTop: '5px' }}>
              <div className="col-md-6">
                {T.translate('transaction.item.confirmations')}: {confirmations}
              </div>
              <div className="col-md-6">
                {T.translate('transaction.item.amount')}:{' '}
                {XVGformatter.format(amount)} XVG
              </div>
            </div>
            <div className="row">
              <div className="col-md-6">
                {T.translate('transaction.item.time')}:{' '}
                {moment.unix(time).format('MM/DD/YYYY hh:mm a')}
              </div>
              <div className="col-md-6">
                {T.translate('transaction.item.timereceived')}:{' '}
                {moment.unix(timereceived).format('MM/DD/YYYY hh:mm a')}
              </div>
            </div>
          </div>
        ) : null}
      </div>
    )
  }
}
