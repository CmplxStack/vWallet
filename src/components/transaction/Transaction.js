import React, { Component } from 'react'
import { inject, observer } from 'mobx-react'
import styled, { keyframes } from 'styled-components'

import ArrowDown from '../../icons/ArrowDown'
import ArrowPop from '../../icons/ArrowPop'
import ArrowUp from '../../icons/ArrowUp'
import BlockChain from '../../icons/Blockchain'
import CheckBadge from '../../icons/CheckBadge'
import PropTypes from 'prop-types'
import T from 'i18n-react'
import Timer from '../../icons/Timer'
import { fadeIn } from 'react-animations'
import { isNull } from 'util'
import moment from 'moment'

const TextContainer = styled.div`
  color: ${props => (props.theme.light ? '#999999;' : '#7193ae;')};
`

const TransactionIcon = styled.div`
  display: inline-flex;
  padding: 7px;
  border-radius: 52%;
  height: 32px;
  width: 32px;
  background-color: ${props => (!props.up ? '#00917a' : '#dc2b3d')};
  .arrow-down,
  .arrow-up {
    stroke-width: 3px;
  }
`

const CenterDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

const NewSign = styled.div`
  border: 1px solid #7193ae;
  border-radius: 1.4em;
  text-align: center;
  text-transform: uppercase;
  font-size: 0.7em;
  height: auto;
  width: 90px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`
const fadeInAnimation = keyframes`${fadeIn}`

const ContainerClicky = styled.div`
  cursor: pointer;
  animation: 1s ${fadeInAnimation};
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.18);
  border-radius: 5px 5px;
`

const TransactionDetails = styled.div`
  animation: 1s ${fadeInAnimation};
  padding-left: 8px;
  margin: 10px 0px;
`

const TransactionDetailsHeader = styled.div`
  font-weight: 800;
  font-size: 14px;
  font-style: bold;
`
const TransactionDetailsFooter = styled.div`
  font-weight: 800;
  font-size: 12px;
`

const TransactionDetailsMoney = styled.div`
  margin-top: 2px;
  margin-bottom: 2px;
  font-weight: 800;
  font-size: 18px;
  color: #000;
`

const SubTransactionDetails = styled.div`
  margin-top: 5px;
  padding-top: 5px;
  border-top: 1px solid rgba(100, 100, 100, 0.07);
  padding-left: 23px;
  padding-right: 15px;
  margin-left: -23px;
  margin-right: -15px;
  text-align: left;
`

const SubTransactionFurtherDetails = styled.div`
  margin-top: 5px;
  padding-top: 5px;
  border-top: 1px solid rgba(100, 100, 100, 0.07);
  padding-left: 23px;
  padding-right: 15px;
  margin-left: -23px;
  margin-right: -15px;
  padding-bottom: 4px;
  text-align: left;
  background-color: rgba(100, 100, 100, 0.07);
  margin-bottom: -10px;
  border-bottom-left-radius: 7px;
  border-bottom-right-radius: 7px;
`

const TransactionDetailProp = styled.div`
  padding: 0px !important;
  display: inline-flex;
  align-items: center;
`

const ExternalLinks = styled.a`
  margin-right: 5px;
  margin-left: 5px;
`

const RoundedTransaction = styled.div`
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  padding: 6px 10px;
  border-bottom: 1px solid rgba(100, 100, 100, 0.07);
  &:hover {
    background-color: hsla(207, 48%, 95%, 0.8);
  }
`
class Transaction extends Component {
  getType(amount, category, fee) {
    if (amount != 0) {
      return category.includes('receive')
        ? T.translate('transaction.item.receive')
        : T.translate('transaction.item.sent')
    } else if ((!amount || amount == 0) && fee < 0) {
      return T.translate('transaction.item.fee')
    }
  }

  isNew() {
    return this.props.timereceived + 90 * 60 - moment().unix() > 0
  }

  render() {
    const XVGformatter = new Intl.NumberFormat(
      this.props.SettingsStore.getLocale,
      {
        minimumSignificantDigits: 1,
        minimumFractionDigits: 2,
        maximumFractionDigits: 4,
      }
    )

    const {
      address = '',
      amount = 0,
      fee = 0,
      blockhash = '',
      category = '',
      confirmations = 0,
      time = 0,
      timereceived = 0,
      txid = '',
      hide = false,
      TransactionStore,
      SettingsStore,
    } = this.props

    return (
      <ContainerClicky
        className="container"
        onClick={() => {
          TransactionStore.setVisibility(txid, category, address, !hide)
        }}
      >
        <RoundedTransaction className="row">
          <div
            className="col-md-1"
            style={{
              textAlign: 'center',
              fontWeight: '500',
              fontSize: '13px',
              paddingTop: '5px',
            }}
          >
            <TextContainer>
              {moment
                .unix(timereceived)
                .format('MMM')
                .toUpperCase()}
            </TextContainer>
            <TextContainer
              style={{
                fontWeight: '300',
                fontSize: '22px',
                lineHeight: '1.0',
                color: '#cacaca',
              }}
            >
              {moment.unix(timereceived).format('DD')}
            </TextContainer>
          </div>
          <CenterDiv className="col-md-1">
            {category.includes('receive') ? (
              <TransactionIcon>
                <ArrowUp width={18} height={18} />
              </TransactionIcon>
            ) : (
              <TransactionIcon up>
                <ArrowDown width={18} height={18} />
              </TransactionIcon>
            )}
          </CenterDiv>
          {this.isNew() ? <TextContainer className="col-md-1" /> : isNull}
          <div
            className={this.isNew() ? 'col-md-8' : 'col-md-9'}
            style={{
              fontWeight: 'bold',
              color: category.includes('receive') ? '#00917a' : '#dc2b3d',
              textAlign: 'right',
              letterSpacing: '1px',
              fontSize: '22px',
            }}
          >
            <div>
              <font
                style={{
                  fontSize: '20px',
                  fontWeight: 400,
                }}
              >
                {Math.abs(amount)
                  .toFixed(3)
                  .toLocaleString(SettingsStore.getLocale)}{' '}
                {category.includes('receive') ? '+' : '-'}
              </font>
            </div>
            <TextContainer
              style={{
                fontSize: '12px',
                fontWeight: '500',
                letterSpacing: '1px',
              }}
            >
              <font>{this.getType(amount, category, fee)}</font>
            </TextContainer>
          </div>
          {blockhash ? (
            <div
              className="col-md-1"
              style={{
                textAlign: 'center',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
              }}
            >
              <ArrowPop height={36} hide={hide} />
            </div>
          ) : (
            <div
              className="col-md-2"
              style={{ textAlign: 'center', marginTop: '15px' }}
            >
              {' '}
              <TextContainer style={{ fontSize: '10px' }}>
                Out of sync
              </TextContainer>
            </div>
          )}
        </RoundedTransaction>
        {!hide ? (
          <TransactionDetails className="trans-details">
            <TransactionDetailsHeader className="Row">
              {this.getType(amount, category, fee)} transaction{' · '}
              {moment.unix(timereceived).fromNow()}
            </TransactionDetailsHeader>
            <TransactionDetailsMoney className="Row">
              XVG {XVGformatter.format(Math.abs(amount))}{' '}
              {category.includes('receive') ? '+' : '-'}
            </TransactionDetailsMoney>
            <TransactionDetailsFooter className="Row">
              {category.includes('receive') ? 'to' : 'from'} {address}
            </TransactionDetailsFooter>
            <SubTransactionDetails className="Row">
              <TransactionDetailProp className="col-md-6">
                <CheckBadge
                  height={15}
                  width={15}
                  style={{ fill: 'rgba(100,100,100, 0.5)', marginRight: '7px' }}
                />{' '}
                {confirmations} confirmations
              </TransactionDetailProp>
              <TransactionDetailProp className="col-md-6">
                <Timer
                  height={15}
                  width={15}
                  style={{
                    fill: 'rgba(100,100,100, 0.75)',
                    marginRight: '7px',
                  }}
                />{' '}
                {moment.unix(timereceived).format('MMMM Do YYYY, h:mm:ss a')}
              </TransactionDetailProp>
            </SubTransactionDetails>
            <SubTransactionFurtherDetails className="Row">
              <TransactionDetailProp className="col-md-12">
                <BlockChain
                  height={15}
                  width={15}
                  style={{ fill: 'rgba(100,100,100, 0.7)', marginRight: '7px' }}
                />{' '}
                Further Information:
                <ExternalLinks href="#">Open Transaction</ExternalLinks>
                {' · '}
                <ExternalLinks href="#">Open Block</ExternalLinks>
              </TransactionDetailProp>
            </SubTransactionFurtherDetails>
          </TransactionDetails>
        ) : null}
      </ContainerClicky>
    )
  }
}

Transaction.propTypes = {
  amount: PropTypes.number.isRequired,
  account: PropTypes.string,
  address: PropTypes.string,
  fee: PropTypes.string,
  blockhash: PropTypes.string,
  category: PropTypes.string,
  confirmations: PropTypes.number,
  time: PropTypes.number,
  timereceived: PropTypes.number,
  txid: PropTypes.string,
  hide: PropTypes.bool.isRequired,
  TransactionStore: PropTypes.object.isRequired,
  SettingsStore: PropTypes.object.isRequired,
}

export default inject('TransactionStore', 'SettingsStore')(
  observer(Transaction)
)

/*OLD TABLE
<div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.address')}:
              </div>{' '}
              <a
                className="col-md-10"
                href="#"
                onClick={() =>
                  shell.openExternal(
                    `https://verge-blockchain.info/address/${address}`
                  )
                }
              >
                {address}
              </a>
            </div>
            <div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.blockhash')}:
              </div>
              <div className="col-md-10">
                {' '}
                <a
                  href="#"
                  onClick={() =>
                    shell.openExternal(
                      `https://verge-blockchain.info/block/${blockhash}`
                    )
                  }
                >
                  <em>{blockhash.substring(0, 40)}...</em>
                </a>
              </div>
            </div>
            <div className="row">
              <div className="col-md-2" style={{ fontWeight: 'bold' }}>
                {T.translate('transaction.item.txid')}:
              </div>
              <div className="col-md-10">
                {' '}
                <a
                  href="#"
                  onClick={() =>
                    shell.openExternal(
                      `https://verge-blockchain.info/tx/${txid}`
                    )
                  }
                >
                  <em>{txid.substring(0, 40)}...</em>
                </a>
              </div>
            </div>

            <div
              className="row"
              style={{
                marginTop: '5px',
              }}
            >
              <div className="col-md-6">
                {T.translate('transaction.item.confirmations')}: {confirmations}
              </div>
              <div className="col-md-6">
                {T.translate('transaction.item.amount')}:{' '}
                {XVGformatter.format(amount)} XVG
              </div>
            </div>
            <div className="row">
              <div className="col-md-6">
                {T.translate('transaction.item.time')}:{' '}
                {moment.unix(time).format('MM/DD/YYYY hh:mm a')}
              </div>
              <div className="col-md-6">
                {T.translate('transaction.item.timereceived')}:{' '}
                {moment.unix(timereceived).format('MM/DD/YYYY hh:mm a')}
              </div>
            </div>
            */
