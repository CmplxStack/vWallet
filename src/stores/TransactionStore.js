import { observable, action, computed } from 'mobx'
import { Client } from 'verge-node-typescript'

class TransactionStore {
	@observable transactions = new Map()

	@action
	addTransaction = transaction => {
		this.transactions.set(transaction.txid, transaction)
	}

	@action
	addTransactions = transactions => {
		transactions.forEach(transaction => {
			this.transactions.set(transaction.txid, transaction)
		})
	}

	@computed
	get getTransactionCount() {
		return this.transactions.size
	}

	@computed
	get getTransactionList() {
		return Array.from(this.transactions.values())
	}
}

const store = new TransactionStore()

const client = new Client({ user: 'kyon', pass: 'lolcat' })

client.getTransactionList().then(transactions => {
	store.addTransactions(transactions)
})

setInterval(() => {
	client.getTransactionList().then(transactions => {
		store.addTransactions(transactions)
	})
}, 5000)

export default store
