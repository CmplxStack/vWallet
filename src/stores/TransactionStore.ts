import * as moment from 'moment';

import { action, computed, decorate, observable } from 'mobx';

import { Client } from 'verge-node-typescript';
import { Transaction } from 'verge-node-typescript/dist/Transaction';

const hash = (transaction: TransactionView) =>
  `${transaction.txid}#${transaction.category}#${transaction.address}#${
    transaction.timereceived
  }`;

const client = new Client({ user: 'kyon', pass: 'lolcat' });

interface TransactionView extends Transaction {
  hide?: boolean;
  fee?: number;
}

export class TransactionStore {
  transactions: Map<string, TransactionView> = new Map();
  loadingFinished: boolean = false;
  search: string = '';

  addTransactions = (transactions: Transaction[] = []) => {
    transactions.forEach((transaction: Transaction) => {
      const oldTransaction = this.transactions.get(hash(transaction));
      this.transactions.set(hash(transaction), {
        hide: true,
        ...oldTransaction,
        ...transaction,
      });
    });
    this.loadingFinished = true;
  };

  setVisibility(txid, category, address, timereceived, hide) {
    const searchedTransaction: Transaction = <Transaction>{
      txid,
      category,
      address,
      timereceived,
    };

    const oldTransaction: TransactionView | undefined = this.transactions.get(
      hash(searchedTransaction),
    );

    if (oldTransaction == null) return;

    const newTransaction: Transaction = <Transaction>{
      txid,
      category,
      address,
      timereceived,
    };

    const updatedTransaction: TransactionView = {
      ...oldTransaction,
      hide,
    };

    this.transactions.set(hash(newTransaction), updatedTransaction);

    this.loadingFinished = true;
  }

  setSearch(e) {
    this.search = e.target.value;
  }

  get getTransactionCount() {
    return this.transactions.size;
  }

  get loaded() {
    return this.loadingFinished;
  }

  get getTransactionList() {
    return this.transactions;
  }

  get searchValue() {
    return this.search;
  }

  get lastTenTransaction() {
    const transactions = Array.from(this.transactions.values());

    if (this.search) {
      return transactions
        .filter(transaction =>
          [
            transaction.address,
            transaction.amount,
            transaction.account,
            transaction.category,
          ]
            .join('-')
            .toLocaleLowerCase()
            .includes(this.search.toLocaleLowerCase()),
        )
        .sort((a: Transaction, b: TransactionView) => b.time - a.time)
        .slice(0, 9);
    }

    return transactions.sort((a, b) => b.time - a.time).slice(0, 9);
  }

  get monthlyOutput() {
    return Array.from(this.transactions.values())
      .filter(
        ({ timereceived, category }) =>
          moment.unix(timereceived).isSame(new Date(), 'month') &&
          category.includes('send'),
      )
      .reduce(
        (sum, { amount, fee }) => (fee ? sum + amount + fee : sum + amount),
        0.0,
      );
  }

  get monthlyIncome() {
    return Array.from(this.transactions.values())
      .filter(
        ({ timereceived, category }) =>
          moment.unix(timereceived).isSame(new Date(), 'month') &&
          category.includes('receive'),
      )
      .reduce((sum, { amount }) => sum + amount, 0.0);
  }
}

decorate(TransactionStore, {
  transactions: observable,
  loadingFinished: observable,
  search: observable,
  addTransactions: action,
  setVisibility: action,
  setSearch: action,
  getTransactionCount: computed,
  loaded: computed,
  getTransactionList: computed,
  searchValue: computed,
  lastTenTransaction: computed,
  monthlyOutput: computed,
  monthlyIncome: computed,
});

const store = new TransactionStore();

client.getTransactionList(100).then(transactions => {
  store.addTransactions(transactions);
});

setInterval(() => {
  client.getTransactionList(100).then(transactions => {
    store.addTransactions(transactions);
  });
}, 10000);

export default store;
