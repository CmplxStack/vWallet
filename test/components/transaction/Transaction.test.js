import React from 'react'
import Transaction from '../../../src/components/transaction/Transaction.js'
import TransactionStore from '../../../src/stores/TransactionStore'
import renderer from 'react-test-renderer'

test('Transaction changes the class when hovered', () => {
  const component = renderer.create(
    <Transaction
      TransactionStore={TransactionStore}
      SettingsStore={{ getLocale: 'en-US' }}
      amount={123.123}
      hide={true}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
