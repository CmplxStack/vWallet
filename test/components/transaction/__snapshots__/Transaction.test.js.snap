// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction changes the class when hovered 1`] = `
<div
  className="container sc-EHOje kqqlKE"
  onClick={[Function]}
>
  <div
    className="row sc-gzVnrw czxydJ"
  >
    <div
      className="col-md-1"
      style={
        Object {
          "fontSize": "13px",
          "fontWeight": "500",
          "paddingTop": "5px",
          "textAlign": "center",
        }
      }
    >
      <div
        className="sc-bwzfXH bFTjyN"
      >
        JAN
      </div>
      <div
        className="sc-bwzfXH bFTjyN"
        style={
          Object {
            "color": "#cacaca",
            "fontSize": "22px",
            "fontWeight": "300",
            "lineHeight": "1.0",
          }
        }
      >
        01
      </div>
    </div>
    <div
      className="col-md-1 sc-bxivhb imfAaG"
    >
      <div
        className="sc-htpNat gVAJuF"
      >
        <svg
          height={18}
          id="Lag_1"
          style={
            Object {
              "enableBackground": "new 0 0 24 24",
            }
          }
          version={1.1}
          viewBox="0 0 24 24"
          width={18}
          x="0px"
          xmlSpace="preserve"
          xmlnsXlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <line
            className="arrow-down"
            x1={12}
            x2={12}
            y1={5}
            y2={19}
          />
          <polyline
            className="arrow-down"
            points="19,12 12,19 5,12 "
          />
        </svg>
      </div>
    </div>
    <div
      className="col-md-9"
      style={
        Object {
          "color": "#dc2b3d",
          "fontSize": "22px",
          "fontWeight": "bold",
          "letterSpacing": "1px",
          "textAlign": "right",
        }
      }
    >
      <div>
        <font
          style={
            Object {
              "fontSize": "10px",
              "fontWeight": 500,
              "verticalAlign": "super",
            }
          }
        >
          XVG
        </font>
        <font
          style={
            Object {
              "fontSize": "20px",
              "fontWeight": 400,
            }
          }
        >
          123.12
           
          -
        </font>
      </div>
      <div
        className="sc-bwzfXH bFTjyN"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "500",
            "letterSpacing": "1px",
          }
        }
      >
        <font>
          transaction.item.sent
        </font>
      </div>
    </div>
    <div
      className="col-md-2"
      style={
        Object {
          "marginTop": "15px",
          "textAlign": "center",
        }
      }
    >
       
      <div
        className="sc-bwzfXH bFTjyN"
        style={
          Object {
            "fontSize": "10px",
          }
        }
      >
        Out of sync
      </div>
    </div>
  </div>
</div>
`;
